extension/healthcheckextension/healthcheckextension.go

```diff
@@ -41,8 +39,7 @@ func (hc *healthCheckExtension) Start(_ context.Context, host component.Host) er
 	hc.logger.Info("Starting health_check extension", zap.Any("config", hc.config))
 
 	// Initialize listener
-	portStr := ":" + strconv.Itoa(int(hc.config.Port))
-	ln, err := net.Listen("tcp", portStr)
+	ln, err := hc.config.TCPAddr.Listen()
```

Comment:
I think we need to do some backwards compatible things here correct?

`if port != 0 and endpoint == default then endpoint = ":" + port`